jetty-keystore/src/main/java/org/eclipse/jetty/keystore/KeystoreGenerator.java

```diff
@@ -0,0 +1,83 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.keystore;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.KeyStore;
+import java.security.SecureRandom;
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Date;
+
+import org.bouncycastle.asn1.x500.X500Name;
+import org.bouncycastle.cert.X509v3CertificateBuilder;
+import org.bouncycastle.cert.jcajce.JcaX509CertificateConverter;
+import org.bouncycastle.cert.jcajce.JcaX509v3CertificateBuilder;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.operator.ContentSigner;
+import org.bouncycastle.operator.jcajce.JcaContentSignerBuilder;
+import org.eclipse.jetty.util.security.Password;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KeystoreGenerator
+{
+    private static final Logger LOG = LoggerFactory.getLogger(KeystoreGenerator.class);
+
+    @SuppressWarnings("unused")
+    public static File generateTestKeystore(String location, String password) throws Exception
+    {
+        LOG.warn("Generating Test Keystore: DO NOT USE IN PRODUCTION!");
```

Comment:
I was thinking that this warning should be called from the XML.
Somebody might use this class outside of the module, so it should not produce the warning.
Also add a info with a link to the documentation about how to generate a real keystore

Reply:
The method is called "generateTestKeystore".
I think it's fair to log a warning here, like this, even for embedded users.

Reply:
Ah yes,
but perhaps we need a `generateKeystore(String algorithm, int size, Instant expires, String cname, String algorithm, String password, File keystore)` method that could be called to make a none test one without a warning?

Reply:
I think we should avoid adding the extra method unless we have some use for it. Would `generateKeystore` be useful to generate a non-test keystore? it would still just be using a self signed certificate.